uk.co.compucorp.civicrm.tasksassignments/tasksassignments.php

```diff
@@ -510,3 +512,37 @@ function _tasksAssignments_allow_only_add_timeline_action (phpQueryObject $doc)
   $actionDropdown->after($addTimelineBtn);
   $actionDropdown->remove();
 }
+
+/**
+ * Remove columns from timeline
+ *
+ * @param phpQueryObject $doc
+ */
+function _tasksAssignments_remove_columns_from_timeline ($doc) {
+  $selectColumnHeader = $doc->find('table th:nth-child(6)');
+  $selectColumnBody = $doc->find('table td:nth-child(6)');
+  $referenceColumnHeader = $doc->find('table th:nth-child(4)');
+  $referenceColumnBody = $doc->find('table td:nth-child(4)');
+  $statusColumnHeader = $doc->find('table th:nth-child(3)');
+  $statusColumnBody = $doc->find('table td:nth-child(3)');
+
+  $elementsToRemove = [
+    $selectColumnHeader, $selectColumnBody,
+    $referenceColumnHeader, $referenceColumnBody,
+    $statusColumnHeader, $statusColumnBody
+  ];
+
+  foreach( $elementsToRemove as $element ) {
```

Comment:
can we refactor this function a bit?

```php
function _tasksAssignments_remove_column_from_timeline ($doc, columnName) {
  $columnHeader = $doc->find('table th:contains("' + columnName + "');
  $column = $doc->find('table td:nth-child(' + $columnHeader->index() + ')');

  $columnHeader->remove();
  $column->remove();
}
```

And use it either like this:
```php
$changeSet->alterHtml('~/crmCaseType/timelineTable.html', function (phpQueryObject $doc) {
  // ...
 _tasksAssignments_remove_column_from_timeline($doc, 'Status');
 _tasksAssignments_remove_column_from_timeline($doc, 'Reference');
 _tasksAssignments_remove_column_from_timeline($doc, 'Select');
});
```

Or like this:
```php
$changeSet->alterHtml('~/crmCaseType/timelineTable.html', function (phpQueryObject $doc) {
  // ...
  $columnsToBeRemovedFromTimeline = [ 'Status', 'Reference', 'Select' ];

  foreach ($columnsToBeRemovedFromTimeline as $columnName) {
    _tasksAssignments_remove_column_from_timeline($doc, $columnName);
  }
});
```

Reply:
The second option seems more preferable, or better still, the initial function name `_tasksAssignments_remove_columns_from_timeline` can be retained while the loop is done within the function for the columns that need to be removed. This will help keeping the hook uncluttered with unnecessary logic.